.github/workflows/build.yml

```diff
@@ -0,0 +1,36 @@
+name: Build
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Maven packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```

Comment:
Also, I seem to remember us having problems (probably months ago) with `SONAR_TOKEN` vs. `SONAR_LOGIN`, etc. I think it was maybe in one of our GitLab repos...maybe in elucidation. Do you remember any of this? Searching the web for `SONAR_TOKEN` doesn't help much. I found this: https://dev.to/remast/using-sonarcloud-with-github-actions-and-maven-31kg

It uses `SONAR_TOKEN` but passes it to the `mvn` command using `-Dsonar.login=$SONAR_TOKEN` but the command here does _not_ do that so not sure how it works.

Reply:
That was with Gitlab and elucidation and their variables.  This is how Sonar said to hook it up and the PR ran against Sonar successfully.